### PR TITLE
Finish edit event functionality

### PR DESCRIPTION
**Pull request by Sarah-Addo**
Accomplishes:
- Getting correct moderator and maintainer status for events
- Completes full edit functionality for events

Screenshot: https://screenshot.googleplex.com/PQ7feWxUH2v

### CODE REVIEW

**Review by jnnybk — APPROVED** [review]
lgtm : )

File: src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    GivrUser user = GivrUser.getCurrentLoggedInUser();
+
+    if (!user.isLoggedIn()) {
+      throw new IllegalArgumentException("ERROR: Unable to register event if user is not logged in.");
+    }
+
+    // Only Moderators can add Events.
+    if (!user.isModeratorOfAnyOrg() && !user.isMaintainer()) {
```
Comment (belraquib) [inline src/main/java/com/google/sps/servlets/EditEventServlet.java]: should this check that the user is an editor of _this_ org?

**Review by belraquib — CHANGES_REQUESTED** [review]

**Review by andrewm898 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/EditEventServlet.java (hunk 1 above)
Comment (Sarah-Addo) [reply src/main/java/com/google/sps/servlets/EditEventServlet.java]: the correct check was actually already in line 67 of event updater so i just removed this check from the servlet

File: src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    EventUpdater eventUpdater = new EventUpdater(eventEntity);
+    long millisecondsSinceEpoch = Instant.now().toEpochMilli();
+    HistoryManager history = new HistoryManager();
+    EmbeddedEntity historyUpdate = history.recordHistory("Event was edited.", millisecondsSinceEpoch);
+
+    // Update rest of Event properties from request form.
+    try {
+      eventUpdater.updateEvent(request, user, /* forRegistration */ false, historyUpdate);
```
Comment (belraquib) [inline src/main/java/com/google/sps/servlets/EditEventServlet.java:66]: Can you verify and add a comment that this checks to see if the user is allowed to update the event?

File: src/main/webapp/event.js [hunk 3]
```diff
@@ -364,20 +376,56 @@ class Event {
...
   }
+
+  parseDate(dateAndHours) {
+    // Will change format from MM DD, YYYY 00:00AM to MM DD, YYYY
+    const date = dateAndHours.propertyMap.date.slice(0, -12);
+    return date;
+  }
+
+  parseDateForValue(dateAndHours) {
+    // Will translate a date in format MM DD, YYYY to YYYY-MM-DD for purpose of prefilling input date value
```
Comment (belraquib) [inline src/main/webapp/event.js:403]: Oof...  I would normally recommend using https://momentjs.com/  for this (and if you want to try, go for it!)

Comment (belraquib) [inline src/main/webapp/event.js:403]: leave a comment that this is the format from the java representation of a Date

File: src/main/webapp/open_hours.js [hunk 4]
```diff
@@ -100,7 +100,13 @@ class TimeOption {
 
     // for edit page get the number of pairs of time input options to preset the values
     let numPairs = 0;
-    if (!forRegistration && this.organizationDay.propertyMap.isOpen) {
+    if (!forRegistration && !this.showOpenClosedOptions) {
+      numPairs = this.organizationDay.propertyMap.fromToPairs.value.length;
+
+      // set initial from to pair, if organization is open there is at least one pair
+      this.dayFromInput.setAttribute("value", this.organizationDay.propertyMap.fromToPairs.value[0].propertyMap.from);
```
Comment (belraquib) [inline src/main/webapp/open_hours.js]: can you store propertyMap into a temp variable for readability?

**Review by belraquib — CHANGES_REQUESTED** [review]
mostly nits

File: src/main/webapp/event.js (hunk 3 above)
Comment (Sarah-Addo) [reply src/main/webapp/event.js:403]: I think I am going to keep it as is for now for the sake of time but I will definitely use it in the future!

**Review by belraquib — APPROVED** [review]

**Pull request merged**